dev/source-release.sh

```diff
@@ -20,72 +20,137 @@
 
 set -e
 
-if [ -z "$1" ]; then
-  echo "Usage: $0 <version-num> <rc-num>"
-  exit
-fi
-
-if [ -z "$2" ]; then
-  echo "Usage: $0 <version-num> <rc-num>"
-  exit
+usage () {
+    echo "usage: $0 [-k <key_id>] [-r <git_remote>] <version_num> <rc_num>"
+    echo "  -k      Specify signing key. Defaults to \"GPG default key\""
+    echo "  -r      Specify Git remote name. Defaults to \"origin\""
+    echo "  -d      Turn on DEBUG output"
+    exit 1
+}
+
+# Default repository remote name
+remote="origin"
+
+while getopts "k:r:d" opt; do
+  case "${opt}" in
+    k)
```

Comment:
(Non-blocking) Should we allow for `--key-id` etc? I think the syntax is just like this for each case: `-k | --key-id)`?

Reply:
The Bash builtin function `getopts` does not support long option names.

Reply:
Oh that's news to me. Looking into it further, I was thinking of `getopt`. It's better to stick with the default installation for most systems though 👍.